### ISSUE TITLE
erase XBM memory after n iteration  + set warmup epochs

### ISSUE THREAD

**Issue by farahFif** [KevinMusgrave/pytorch-metric-learning#330]
Hello, 

1 - Is there a way to erase the XBM memory ? if the XBM loss is called at each new iteration does it ensure that the memory with old embeddings is erased ? 

2 - Is there a way to set warmup epochs as suggested in the paper ? 

Thanks

**Comment by KevinMusgrave** [comment]
> 1 - Is there a way to erase the XBM memory ? if the XBM loss is called at each new iteration does it ensure that the memory with old embeddings is erased ?

Yes, once the queue is filled, the oldest embeddings are replaced with the newest embeddings at each iteration. Currently there isn't a function for deleting everything in the queue, but you could use this function as a workaround:
```python
def reset_xbm_queue(loss_fn):
        loss_fn.embedding_memory = torch.zeros(loss_fn.memory_size, loss_fn.embedding_size)
        loss_fn.label_memory = torch.zeros(loss_fn.memory_size).long()
        loss_fn.has_been_filled = False
        loss_fn.queue_idx = 0
```

 
> 2 - Is there a way to set warmup epochs as suggested in the paper ?
I guess during warmup they simply don't use the memory queue? This feature currently isn't built into CrossBatchMemory.

**Comment by KevinMusgrave** [comment]
Added this to v1.0.0:
```python
loss_fn.reset_queue()
```
